cmd/dns_add.go

```diff
@@ -20,28 +19,79 @@ func init() {
 	dnsCmd.AddCommand(dnsAddCmd)
 }
 
+// Create a map to store the string to CreateDNSDomainRecordRequestType mappings
+var dnsRecordTypeMap = map[string]v3.CreateDNSDomainRecordRequestType{
+	"NS":    v3.CreateDNSDomainRecordRequestTypeNS,
+	"CAA":   v3.CreateDNSDomainRecordRequestTypeCAA,
+	"NAPTR": v3.CreateDNSDomainRecordRequestTypeNAPTR,
+	"POOL":  v3.CreateDNSDomainRecordRequestTypePOOL,
+	"A":     v3.CreateDNSDomainRecordRequestTypeA,
+	"HINFO": v3.CreateDNSDomainRecordRequestTypeHINFO,
+	"CNAME": v3.CreateDNSDomainRecordRequestTypeCNAME,
+	"SSHFP": v3.CreateDNSDomainRecordRequestTypeSSHFP,
+	"SRV":   v3.CreateDNSDomainRecordRequestTypeSRV,
+	"AAAA":  v3.CreateDNSDomainRecordRequestTypeAAAA,
+	"MX":    v3.CreateDNSDomainRecordRequestTypeMX,
+	"TXT":   v3.CreateDNSDomainRecordRequestTypeTXT,
+	"ALIAS": v3.CreateDNSDomainRecordRequestTypeALIAS,
+	"URL":   v3.CreateDNSDomainRecordRequestTypeURL,
+	"SPF":   v3.CreateDNSDomainRecordRequestTypeSPF,
+}
```

Comment:
```
var dnsRecordTypeMap = map[string]v3.CreateDNSDomainRecordRequestType{
	string(v3.CreateDNSDomainRecordRequestTypeNS): v3.CreateDNSDomainRecordRequestTypeNS,
	// and so on for each type
}
```
you could do this instead, leaves less room for spelling errors.

Reply:
We can probably use stringer here in egoscale what do you think @sauterp ?